### DISCUSSION TITLE
Setting the one-time dispatcher

### DISCUSSION THREAD

**Discussion by dglozic** [nodejs/undici#2031] [Q&A] [answered]
I would like to tweak some options but only a the single request. I see that Agent has a constructor and can accept an object with properties (is that correct type to pass in)? Do I set it as a `dispatcher` property in the options? Do I need to do something with it after (dispose of it somehow) to avoid leakage?

**Comment by dglozic** [comment, marked as answer]
I *think* setting the `dispatcher` to `new Agent({<options>})` works, just need confirmation.

**Reply by mcollina** [reply]
confirmed!

**Comment by dglozic** [comment]
As discussed in https://github.com/nodejs/undici/discussions/2031, I will create a shared new Agent that I will use in lieu of the default dispatcher but I will reuse it across requests that share the same need (in this case, handling of self-signed certificates). We do not use these when deployed, this is just to support running in localhost.
